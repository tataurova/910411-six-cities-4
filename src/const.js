export const MapSettings = {
  ICON_URL: `img/pin.svg`,
  ACTIVE_ICON_URL: `img/pin-active.svg`,
  ICON_SIZE: [30, 30],
};

export const CardType = {
  CITY: `cities`,
  NEAR_PLACE: `near-places`,
  FAVORITE: `favorites`,
};

export const SortType = {
  DEFAULT: `default`,
  PRICE_UP: `price-up`,
  PRICE_DOWN: `price-down`,
  RATING_DOWN: `rating-down`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  PLACE_FULL_CARD: `/offer`,
  FAVORITES: `/favorites`,
  FAVORITE: `/favorite`,
  HOTELS: `/hotels`,
  COMMENTS: `/comments`,
  NEARBY: `/nearby`,
  SERVER_URL: `https://4.react.pages.academy/six-cities`,
};

export const LoginInput = {
  EMAIL: `email`,
  PASSWORD: `password`,
};

export const Error = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404,
  SERVICE_UNAVAILABLE: 503,
  TIMEOUT_ERROR: `timeout_error`,
};

export const FavoriteStatus = {
  ADD: 1,
  REMOVE: 0,
};

export const RatingValue = {
  ONE: 1,
  TWO: 2,
  THREE: 3,
  FOUR: 4,
  FIVE: 5,
};

export const CommentLength = {
  START_TYPING: 1,
  MIN: 50,
};

export const SORT_TYPES = [
  `Popular`,
  `Price: low to high`,
  `Price: high to low`,
  `Top rated first`,
];

export const INITIAL_STATE_HOVERED_CARD = 0;

export const NEAR_PLACES_MAX_COUNT = 3;

export const MAP_NEAR_PLACES_MAX_COUNT = 2;

export const DEFAULT_HOVERED_CARD = 0;

export const DEFAULT_SORT_STATE = false;

export const MAX_TIMEOUT = 5000;

export const LOGIN_MIN_LENGTH = 6;

export const PASSWORD_MIN_LENGTH = 1;

export const DEFAULT_RATING = 0;

export const SHOW_REVIEW_MAX_NUMBER = 10;

export const DEFAULT_ERROR_STATUS = false;

export const SUCCESS_RESPONSE_CODE = 200;

export const SHOW_ERROR_TIMEOUT = 3000;

