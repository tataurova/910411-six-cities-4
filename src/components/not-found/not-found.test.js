import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import NotFound from "./not-found.tsx";
import {AuthorizationStatus, DEFAULT_ERROR_STATUS} from "../../const.js";

describe(`<NotFound />`, () => {
  it(`Should NotFound render correctly`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <NotFound
              authorizationStatus = {AuthorizationStatus.NO_AUTH}
              user = {``}
              error = {DEFAULT_ERROR_STATUS}
            />
          </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
