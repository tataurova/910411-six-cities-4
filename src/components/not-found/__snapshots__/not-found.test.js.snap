// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Should NotFound render correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              Not Found
            </b>
            <p
              className="cities__status-description"
            >
              This page does not exist
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
